### PR TITLE
Blazor URI and navigation state helpers

### LINKED ISSUE #11397: Add Programatic Routing with IUriHelper
I know it exists but currently it's not documented how to navigate programatically (IUriHelper) If im not mistaken this used to be documented.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cb7d0b96-7f5c-7330-f111-6e3f8fb94766
* Version Independent ID: 14285806-d00d-d7d9-594c-91f35e2912a0
* Content: [Razor Components routing](https://docs.microsoft.com/en-us/aspnet/core/razor-components/routing?view=aspnetcore-3.0#feedback)
* Content Source: [aspnetcore/razor-components/routing.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/razor-components/routing.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #11397

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/routing?view=aspnetcore-3.0&branch=pr-en-us-12419)

Thanks @vertonghenb! :rocket:

### CODE REVIEW

**Comment by guardrex** [comment]
ping @danroth27 for review

File: aspnetcore/blazor/routing.md [hunk 1]
```diff
@@ -102,3 +102,37 @@ There are two `NavLinkMatch` options:
...
 In the preceding example, the Home NavLink (`href=""`) matches all URLs and always receives the `active` CSS class. The second NavLink only receives the `active` class when the user visits the Blazor Route component (`href="BlazorRoute"`).
+
+## URI and navigation state helpers
+
+Use `Microsoft.AspNetCore.Components.IUriHelper` to work with URIs and navigation in C# code. `IUriHelper` provides the event and methods shown in the following table.
+
+| Member | Description |
+| ------ | ----------- |
+| `GetAbsoluteUri` | Gets the current absolute URI. |
+| `GetBaseUri` | Gets the base URI (with a trailing slash) that can be prepended to relative URI paths to produce an absolute URI. Typically, this corresponds to the `href` attribute on the document's `<base>` element in *wwwroot/index.html* (Blazor client-side) or *Pages/\_Host.cshtml* (Blazor server-side). |
```
Comment (tdykstra) [inline aspnetcore/blazor/routing.md]: "this"  is ambiguous, could refer to base or absolute from previous sentence.

File: aspnetcore/blazor/routing.md [hunk 2]
```diff
@@ -102,3 +102,37 @@ There are two `NavLinkMatch` options:
...
+| `NavigateTo` | Navigates to the specified URI. If `forceLoad` is `true`:<ul><li>Client-side routing is bypassed.</li><li>The browser is forced to load the new page from the server, whether or not the URI is normally handled by the client-side router.</li></ul> |
+| `OnLocationChanged` | An event that fires when the navigation location has changed. |
+| `ToAbsoluteUri` | Converts a relative URI into an absolute URI by resolving the URI relative to the current absolute URI. |
```
Comment (tdykstra) [inline aspnetcore/blazor/routing.md]: The first half and second half of the sentence seem to say the same thing.

File: aspnetcore/blazor/routing.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/blazor/routing.md]: It refers back to the member itself. I agree with u ... I probably cut-'n-paste it from the API ... always a dangerous move from an English perspective. :trollface: :smile: lol

**Pull request merged**